### PR TITLE
HiverServer2 based driver

### PR DESCRIPTION
**Pull request by intamyuto**
* implement hs2 client 
* driver now conforms sql/driver interfaces
* add minimal test suite
* update README

### CODE REVIEW

File: rows.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+// ColumnTypeDatabaseTypeName returns column's database type name
+func (r *Rows) ColumnTypeDatabaseTypeName(index int) string {
+	return r.schema.Columns[index].DatabaseTypeName
+}
+
+// Next prepares next row for scanning
+func (r *Rows) Next(dest []driver.Value) error {
+	return r.rs.Next(dest)
+}
```
Comment (paribhasha) [inline rows.go]: we should also implement the Scan() method for rows - https://golang.org/pkg/database/sql/#Rows.Scan

Comment (intamyuto) [reply rows.go]: we implement driver.Rows Next method so it can be used to provide data for sql.Rows Scan method

Comment (paribhasha) [reply rows.go]: LGTM

**Comment by timarmstrong** [comment]
Wow, very cool

**Comment by vishjosh** [comment]
@timarmstrong we are using it in production now and it is working smooth so far for us. 
Better than previous version!

**Pull request merged**